test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -130,7 +137,123 @@ public void IsWarmUpRequest_ReturnsExpectedValue()
+
+                var loggerProvider = new TestLoggerProvider();
+                var loggerProviderFactory = new TestLoggerProviderFactory(loggerProvider);
+                var webHostSettings = new WebHostSettings
+                {
+                    IsSelfHost = true,
+                    LogPath = Path.Combine(testRootPath, "Logs"),
+                    SecretsPath = Path.Combine(testRootPath, "Secrets"),
+                    ScriptPath = Path.Combine(testRootPath, "WWWRoot")
+                };
+
+                var loggerFactory = new LoggerFactory();
+                loggerFactory.AddProvider(loggerProvider);
+
+                var webHostBuilder = Program.CreateWebHostBuilder()
+                    .ConfigureServices(c => {
+                        c.AddSingleton(webHostSettings)
+                        .AddSingleton<ILoggerProviderFactory>(loggerProviderFactory)
+                        .AddSingleton<ILoggerFactory>(loggerFactory);
+                    });
+
+                var httpServer = new TestServer(webHostBuilder);
+                var httpClient = httpServer.CreateClient();
+                httpClient.BaseAddress = new Uri("https://localhost/");
+
+                TestHelpers.WaitForWebHost(httpClient);
+
+                var traces = loggerProvider.GetAllLogMessages().ToArray();
+                Assert.NotNull(traces.Single(p => p.FormattedMessage.StartsWith("Starting Host (HostId=placeholder-host")));
+                Assert.NotNull(traces.Single(p => p.FormattedMessage.StartsWith("Host is in standby mode")));
+
+                // issue warmup request and verify
+                var request = new HttpRequestMessage(HttpMethod.Get, "api/warmup");
+                var response = await httpClient.SendAsync(request);
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                string responseBody = await response.Content.ReadAsStringAsync();
+                Assert.Equal("WarmUp complete.", responseBody);
+
+                // issue warmup request with restart and verify
+                request = new HttpRequestMessage(HttpMethod.Get, "api/warmup?restart=1");
+                response = await httpClient.SendAsync(request);
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                responseBody = await response.Content.ReadAsStringAsync();
+                Assert.Equal("WarmUp complete.", responseBody);
+
+                // Now specialize the host by invoking assign
+                var secretManager = httpServer.Host.Services.GetService<ISecretManager>();
+                var masterKey = (await secretManager.GetHostSecretsAsync()).MasterKey;
+                string uri = "admin/instance/assign";
+                request = new HttpRequestMessage(HttpMethod.Post, uri);
+                var environment = new Dictionary<string, string>();
+                var assignmentContext = new HostAssignmentContext
+                {
```

Comment:
No functions content on the assign - just the minimum required to cause the specialization. We could add a zip later and verify that as well.